run_tests.py

```diff
@@ -46,13 +46,13 @@
 
 NINJA_1_9_OR_NEWER = False
 NINJA_CMD = None
-# If we're on CI, just assume we have ninja in PATH and it's new enough because
-# we provide that. This avoids having to detect ninja for every subprocess unit
-# test that we run.
-if 'CI' in os.environ:
-    NINJA_1_9_OR_NEWER = True
-    NINJA_CMD = ['ninja']
-else:
+# If we're on CI, detecting ninja for every subprocess unit test that we run is slow
+# Optimize this by respecting $NINJA and skipping detection, then exporting it on
+# first run.
+try:
+    NINJA_1_9_OR_NEWER = bool(int(os.environ['NINJA_1_9_OR_NEWER']))
+    NINJA_CMD = [os.environ['NINJA']]
```

Comment:
Shouldn't these be the other way around? If the first is not set then it throws an exception and the `NINJA` envvar is never read.

Reply:
That's fine, if we throw an exception then we ignore NINJA_CMD anyway so the assignment is meaningless. In the except block we read the environment variable again, using `detect_ninja()`.